fetch_contigs.py

```diff
@@ -0,0 +1,35 @@
+import sys
+import os
+import urllib.request
+from Bio import Entrez
+
+def main(email=None, api_key=None, retmax=None):
+    Entrez.email = email
+    Entrez.api_key = api_key
+
+    handle = Entrez.esearch(db="assembly", term = "E. coli[Organism] AND contig[Assembly Level]", retmax = num)
+
+    record = Entrez.read(handle)
+    assembly_count = 0
+    for i in record['IdList']:
+        handle1 = Entrez.esummary(db="assembly", id = i)

```

Comment:
have you considered doing a batch esummary request. From reading the documentation id can take a list of ids that is comma separated. Maybe it would be good to do this in batchs, i.e. 10. This would give you experience with array subsetting in python.

Reply:
record['IdList'] will contain number of ids specified by the argument. Since I iterate through every item in the list to make a url retrievable key, I dont know how I could do the same operation in batch

Reply:
Didn't realize the objective was to minimize ncbi query attempts. I use esummary only once using join function. Then I use an inner for loop to generate url links and retrieve them one by one. 

handle1 = Entrez.esummary(db="assembly", id = ",".join(record['IdList']))
        record1 = Entrez.read(handle1)
        for i in range(0,int(num)):
            url = record1['DocumentSummarySet']['DocumentSummary'][i]['FtpPath_GenBank']
            file_name = os.path.basename(url)
            link = os.path.join(url,file_name+'_genomic.fna.gz')
            print(f'{file_name}.fna.gz' +" downloaded")
            assembly_count += 1
            urllib.request.urlretrieve(link, f'{file_name}.fna.gz')

        print ("Total {} assemblies downloaded".format(assembly_count))

Reply:
Is there a limit to how many ids ncbi will accept? i.e. if you request 10000 assemblies

Reply:
When I pass 10,000 as an argument, the script starts working, even though it takes a while to download first assembly. Is there a way to know whether it will be successful without waiting till eternity? 

$ python fetch_contigs.py arnab22.iitkgp@gmail.com 9b946150b1df235cf0e7c288ca4588a97c08 E. Coli 10000
In Main
arnab22.iitkgp@gmail.com
9b946150b1df235cf0e7c288ca4588a97c08
Organism is E. Coli
GCA_011292655.1_ASM1129265v1.fna.gz downloaded
GCA_011290775.1_ASM1129077v1.fna.gz downloaded
GCA_003768505.2_PDT000399854.2.fna.gz downloaded
GCA_003912155.2_PDT000394743.2.fna.gz downloaded
GCA_003764045.2_PDT000399320.2.fna.gz downloaded
GCA_003896275.2_PDT000383993.2.fna.gz downloaded
GCA_003754705.2_PDT000399853.2.fna.gz downloaded
GCA_003740885.2_PDT000396585.2.fna.gz downloaded
GCA_003760945.2_PDT000399922.2.fna.gz downloaded
GCA_003736085.2_PDT000396615.2.fna.gz downloaded
GCA_003757625.2_PDT000400034.2.fna.gz downloaded
GCA_003736365.2_PDT000396657.2.fna.gz downloaded

Reply:
If it starts downloading a single file that would tell you that it is fine, could also look at the size of the document returned. otherwise I would refer to there documentation and see if it has a limit.